File: frontend/src/types/wordlist.ts
export interface WordList {
    id: number;
    name: string;
    description: string;
    source: string;
    file_path: string;
    word_count: number;
    created_at: string;
    updated_at: string;
}

export interface WordListResponse {
    word_lists: WordList[];
}

export interface WordListDetailResponse {
    word_list: WordList;
}

export interface WordSampleResponse {
    words: string[];
    count: number;
}

export interface CreateWordListResponse {
    message: string;
    word_list: WordList;
}

export interface UpdateWordListResponse {
    message: string;
    word_list: WordList;
}

export interface DeleteWordListResponse {
    message: string;
}

export interface UseWordListResponse {
    message: string;
}

File: backend/models/wordlist.go
package models

import (
	"time"
)

// WordList represents a list of words with metadata
type WordList struct {
	ID          int       `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Source      string    `json:"source"`
	FilePath    string    `json:"file_path"`
	WordCount   int       `json:"word_count"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
